src/sys/unix/awakener.rs

```diff
@@ -72,3 +77,66 @@ mod pipe {
         }
     }
 }
+
+/// Awakener backed by an eventfd on Linux
+#[cfg(any(target_os = "linux", target_os = "android"))]
+mod eventfd {
+    use sys::unix;
+    use {io, Ready, Poll, PollOpt, Token};
+    use event::Evented;
+    use std::io::{Read, Write};
+    use std::mem::transmute;
+
+    /*
+     *
+     * ===== Awakener =====
+     *
+     */
+
+    pub struct Awakener {
+        fd: unix::Io,
+    }
+
+    impl Awakener {
+        pub fn new() -> io::Result<Awakener> {
+            Ok(Awakener { fd: unix::eventfd()? })
+        }
+
+        pub fn wakeup(&self) -> io::Result<()> {
+            let data: [u8; 8] = unsafe { transmute(1u64) };
+            match (&self.fd).write(&data) {
+                Ok(_) => Ok(()),
+                Err(e) => {
+                    if e.kind() == io::ErrorKind::WouldBlock {
+                        Ok(())
+                    } else {
+                        Err(e)
+                    }
+                }
+            }
+        }
+
+        pub fn cleanup(&self) {
+            let mut buf = [0; 8];
+            let _ = (&self.fd).read(&mut buf);
```

Comment:
This can return an `WouldBlock` error if not calls to `wakeup` were made yet, we might want to hide that error.